zipline/finance/performance/tracker.py

```diff
@@ -385,11 +400,15 @@ def handle_market_close(self, dt, data_portal):
 
             benchmark_value = self.all_benchmark_returns[completed_session]
 
+            portfolio = self.get_portfolio(False)
+            expected_shortfall = portfolio.expected_shortfall
```

Comment:
Do we need to calculate this every day during the backtest (as opposed to just once at the end when we can fetch all the data we need in one call)?  I'm very worried about the performance impact of this change.